x/superfluid/keeper/migrate_test.go

```diff
@@ -0,0 +1,211 @@
+
+			// Add a sanctioned link between the balancer and concentrated liquidity pool.
+			migrationRecord := gammtypes.MigrationRecords{BalancerToConcentratedPoolLinks: []gammtypes.BalancerToConcentratedPoolLink{
+				{BalancerPoolId: pool.GetId(), ClPoolId: clPool.GetId()},
+			}}
+			gammKeeper.SetMigrationInfo(ctx, migrationRecord)
+
+			// The amount of coins we lock is the amount of LP tokens the account that joined the pool has.
+			coinsToLock := poolShareOut
+
+			// The unbonding duration is the same as the staking module's unbonding duration.
+			unbondingDuration := stakingKeeper.GetParams(ctx).UnbondingTime
+
+			// Lock the LP tokens for the duration of the unbonding period.
+			lockID := suite.LockTokens(poolJoinAcc, sdk.NewCoins(coinsToLock), unbondingDuration)
+
+			// Superfluid delegate the lock if the test case requires it.
+			// Note the intermediary account that was created.
+			intermediaryAcc := types.SuperfluidIntermediaryAccount{}
+			if tc.superfluidDelegated {
+				err = superfluidKeeper.SuperfluidDelegate(ctx, poolJoinAcc.String(), lockID, valAddr.String())
+				suite.Require().NoError(err)
+				intermediaryAccConnection := superfluidKeeper.GetLockIdIntermediaryAccountConnection(ctx, lockID)
+				intermediaryAcc = superfluidKeeper.GetIntermediaryAccount(ctx, intermediaryAccConnection)
+			}
+
+			// Superfluid undelegate the lock if the test case requires it.
+			if tc.superfluidUndelegating {
+				err = superfluidKeeper.SuperfluidUndelegate(ctx, poolJoinAcc.String(), lockID)
+				suite.Require().NoError(err)
+			}
+
+			// Unlock the lock if the test case requires it.
+			if tc.unlocking {
+				// If lock was superfluid staked, we can't unlock via `BeginUnlock`,
+				// we need to unlock lock via `SuperfluidUnbondLock`
+				if tc.superfluidUndelegating {
+					err = superfluidKeeper.SuperfluidUnbondLock(ctx, lockID, poolJoinAcc.String())
+					suite.Require().NoError(err)
+				} else {
+					lock, err := lockupKeeper.GetLockByID(ctx, lockID)
+					suite.Require().NoError(err)
+					err = lockupKeeper.BeginUnlock(ctx, lockID, lock.Coins)
+					suite.Require().NoError(err)
+
+					// add time to current time to test lock end time
+					ctx = ctx.WithBlockTime(ctx.BlockTime().Add(time.Hour * 24))
+				}
+			}
+
+			lock, err := lockupKeeper.GetLockByID(ctx, lockID)
+			suite.Require().NoError(err)
+
+			// Run the unlock and migrate logic.
+			amount0, amount1, _, _, _, err := superfluidKeeper.UnlockAndMigrate(ctx, poolJoinAcc, lockID, poolShareOut)
+			suite.Require().NoError(err)
+
+			suite.AssertEventEmitted(ctx, gammtypes.TypeEvtPoolExited, 1)
+
+			// exitPool has rounding difference.
+			// We test if correct amt has been exited and frozen by comparing with rounding tolerance.
+			defaultErrorTolerance := osmomath.ErrTolerance{
+				AdditiveTolerance: sdk.NewDec(2),
```

Comment:
Do you have a brief number of how much the joinPoolAmt is here? (Asking to get a brief sense of the significance of the tolerance here)

Reply:
Yeah, these values are equal to 5000 uosmo so the tolerance would be 2 uosmo compared to that. To be clear though (and this originally wasn't clear to me before) the user doesn't get shorted these 2 uosmo, they simply remain in the wallet and are free from any lock. Even with very large amounts this amounts to dust though.

Reply:
Sweet! Thanks for clarifying